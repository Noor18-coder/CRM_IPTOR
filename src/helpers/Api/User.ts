import axios from 'axios';

import { ApiRequest } from './ApiRequest';
import { get } from 'lodash';
import * as models from './models';
export class User {
    /** API Method */
   //private static apiMethod: string = 'user.get';
    private static apiMethod: string = 'user.get';
   
    /**
     * Helper function to fetch sales order list
     * @param params user id
     * @returns information about user
     */
   static async get(userId?: string): Promise<models.UserResponse> {
    //static async get(): Promise<CompaniesResponse> {
      const requestData = new ApiRequest<models.UserParams>(this.apiMethod, { user: userId });
      //const requestData = new ApiRequest(this.apiMethod);
      //const response = await axios.post<models.UserResponse>('/api/service', requestData);
      //return get(response, 'data.data', []);
      const response = await axios.post<models.UserResponse>('/api/service', requestData);
      return get(response, 'data.data',{});

    }
  
  }
   