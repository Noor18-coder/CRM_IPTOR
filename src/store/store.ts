import { applyMiddleware, combineReducers, createStore, Store, Reducer } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import storageSession from 'redux-persist/lib/storage/session'
import { composeWithDevTools } from 'redux-devtools-extension';

import { authReducer } from './Auth/Reducers';
import opportunityReducer from './Opportunity/Reducers';
import { OpportunityState } from './Opportunity/Types';
import businessPartnerReducer from './Customer/Reducer';
import { BusinessPartnerState } from './Customer/Types';
import { AuthState, AuthActions } from './Auth/Types';
import usersReducer from './Users/Reducers';
import { SaveUserAction, UsersData } from './Users/Types';
import { AppLoadingActions , InitialConfigState } from './InitialConfiguration/Types';
import configReducer from './InitialConfiguration/Reducers';
import {   AddOpportunityState } from './AddOpportunity/Types';
import addOpportunityReducer from './AddOpportunity/Reducers';

const persistConfig = {
  key: 'root',
  storage: storageSession,
  whitelist: ['auth','enviornmentConfigs', 'addOpportunity'] // which reducer want to store
};

export type AppActions = AuthActions;

export interface AppState {
  readonly auth : AuthState,
  readonly users: UsersData,
  readonly opportunities: OpportunityState,
  readonly businesspartners: BusinessPartnerState,
  readonly enviornmentConfigs:InitialConfigState,
  readonly addOpportunity: AddOpportunityState
}
const appReducer = combineReducers({
  auth: authReducer,
  users: usersReducer,
  opportunities: opportunityReducer,
  businesspartners: businessPartnerReducer,
  enviornmentConfigs: configReducer,
  addOpportunity: addOpportunityReducer
});

const rootReducer = (state: any, action: AppActions) => {
  if (action.type === 'LOGOUT_SUCCESS') {
    state = undefined
  }

  return appReducer(state , action)
}

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(persistedReducer, composeWithDevTools(applyMiddleware(thunk)));
const persistor = persistStore(store);

export {
  store, persistor
};