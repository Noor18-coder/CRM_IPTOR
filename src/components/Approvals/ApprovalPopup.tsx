import React from 'react';
import { Dispatch } from 'redux';
import { useSelector, useDispatch } from 'react-redux';
import ImageConfig from '../../config/ImageConfig';
import { AppState } from '../../store/store';

import ApprovalHistory from './ApprovalHistory';
import SubmitForApproval from './SubmitForApproval';
import ApproverSubmit from './ApproverSubmit';

import { openOpportunityForm } from '../../store/OpportunityDetails/Actions';

interface Props {
  reloadOpportunityDetailsPage: () => void;
}
const ApprovalPopup: React.FC<Props> = ({ reloadOpportunityDetailsPage }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const dispatch: Dispatch<any> = useDispatch();
  const [headerName, setHeaderName] = React.useState<string>('Edit Opportunity');

  const closeAction = () => {
    document.body.classList.remove('body-scroll-hidden');
    dispatch(openOpportunityForm({ open: false }));
  };

  const loadComponent = () => {
    const { groupName } = state.opportuntyDetails.editOportunity;

    if (groupName === 'history') {
      return <ApprovalHistory />;
    } else if (groupName === 'submit_approval') {
      return <SubmitForApproval reloadOpportunity={reloadOpportunityDetailsPage} />;
    } else {
      return <ApproverSubmit reloadOpportunity={reloadOpportunityDetailsPage} />;
    }
  };

  React.useEffect(() => {
    const { groupName } = state.opportuntyDetails.editOportunity;
    if (groupName === 'history') {
      setHeaderName('Approvals History');
    } else if (groupName === 'submit_approval') {
      setHeaderName('You Need Approval');
    } else if (groupName === 'approverSubmit') {
      setHeaderName('Approval');
    }
  }, []);

  return (
    <>
      <div className="sliding-panel-container">
        <div className="sliding-panel">
          <div className="title-row opp-header-text">
            <button type="button" className="mob-steps-back" onClick={closeAction}>
              <img src={ImageConfig.CHEVRON_LEFT} alt="Back" />
            </button>
            {headerName}
            <button type="button" className="panel-close-icon link-anchor-button" onClick={closeAction}>
              <img src={ImageConfig.CLOSE_BTN} alt="Close" />
            </button>
          </div>

          <div className="all-opportunity-steps-container">
            <div className="approval-forms">
              {/* <p className="stepone-title">Opportunity Name</p> */}
              <div className="">
                <div className="steps-one-forms">{loadComponent()}</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default ApprovalPopup;
