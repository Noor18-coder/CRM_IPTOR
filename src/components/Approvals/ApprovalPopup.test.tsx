import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ApprovalPopup from './ApprovalPopup';
import StoreMock from '../../mocks/Store.mock';
import { UserMock } from '../../mocks/User.mock';
import { AppState } from '../../store';

const reloadOpportunityDetailsPage = jest.fn();

describe('[Approvals] ApprovalPopup', () => {
  it('should renders correctly', () => {
    const store = StoreMock.createInitialStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <ApprovalPopup reloadOpportunityDetailsPage={reloadOpportunityDetailsPage} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render component to show approval popup history values', () => {
    const initialState = StoreMock.createAppInitialState();
    const state: AppState = {
      ...initialState,
      users: { ...initialState.users, users: UserMock.getUsers(1) },
      opportuntyDetails: {
        ...initialState.opportuntyDetails,
        editOportunity: { ...initialState.opportuntyDetails.editOportunity, groupName: 'history' },
      },
    };

    const store = StoreMock.createAnyStore<AppState>(state);

    const tree = renderer
      .create(
        <Provider store={store}>
          <ApprovalPopup reloadOpportunityDetailsPage={reloadOpportunityDetailsPage} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render component to show approval popup submit approval values', () => {
    const initialState = StoreMock.createAppInitialState();
    const state: AppState = {
      ...initialState,
      users: { ...initialState.users, users: UserMock.getUsers(1) },
      opportuntyDetails: {
        ...initialState.opportuntyDetails,
        editOportunity: { ...initialState.opportuntyDetails.editOportunity, groupName: 'submit_approval' },
      },
    };

    const store = StoreMock.createAnyStore<AppState>(state);

    const tree = renderer
      .create(
        <Provider store={store}>
          <ApprovalPopup reloadOpportunityDetailsPage={reloadOpportunityDetailsPage} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render component to show approver submit values', () => {
    const initialState = StoreMock.createAppInitialState();
    const state: AppState = {
      ...initialState,
      users: { ...initialState.users, users: UserMock.getUsers(1) },
      opportuntyDetails: {
        ...initialState.opportuntyDetails,
        editOportunity: { ...initialState.opportuntyDetails.editOportunity, groupName: 'approverSubmit' },
      },
    };

    const store = StoreMock.createAnyStore<AppState>(state);

    const tree = renderer
      .create(
        <Provider store={store}>
          <ApprovalPopup reloadOpportunityDetailsPage={reloadOpportunityDetailsPage} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
