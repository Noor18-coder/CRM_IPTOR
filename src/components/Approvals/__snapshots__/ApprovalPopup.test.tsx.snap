// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Approvals] ApprovalPopup should render component to show approval popup history values 1`] = `
<div
  className="sliding-panel-container"
>
  <div
    className="sliding-panel"
  >
    <div
      className="title-row opp-header-text"
    >
      <button
        className="mob-steps-back"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Back"
          src="chevron-left-black.svg"
        />
      </button>
      Edit Opportunity
      <button
        className="panel-close-icon link-anchor-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Close"
          src="close.svg"
        />
      </button>
    </div>
    <div
      className="all-opportunity-steps-container"
    >
      <div
        className="opportunity-forms"
      >
        <div
          className=""
        >
          <div
            className="steps-one-forms"
          >
            <div
              className="stage-level-container"
            >
              <div
                className="stage-level-section"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Approvals] ApprovalPopup should render component to show approval popup submit approval values 1`] = `
<div
  className="sliding-panel-container"
>
  <div
    className="sliding-panel"
  >
    <div
      className="title-row opp-header-text"
    >
      <button
        className="mob-steps-back"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Back"
          src="chevron-left-black.svg"
        />
      </button>
      You Need Approval
      <button
        className="panel-close-icon link-anchor-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Close"
          src="close.svg"
        />
      </button>
    </div>
    <div
      className="all-opportunity-steps-container"
    >
      <div
        className="opportunity-forms"
      >
        <div
          className=""
        >
          <div
            className="steps-one-forms"
          >
            <div
              className="approval-container"
            >
              <p
                className="lbl-apporver-name"
              >
                Approverâ€™s Name
              </p>
              <p
                className="apporver-name"
              >
                <div
                  className="approver-search-container"
                >
                  <input
                    className="form-control"
                    disabled={true}
                    type="text"
                    value=""
                  />
                  <button
                    className="edit-approver-name"
                    onClick={[Function]}
                    type="button"
                  >
                    <img
                      alt="edit"
                      src="edit-icon.svg"
                    />
                  </button>
                </div>
              </p>
              <div
                className="form-group oppty-form-elements"
                id="comment"
              >
                <label
                  className="opp-label"
                  htmlFor="comments"
                >
                  Comment
                </label>
                <textarea
                  className="form-control"
                  onChange={[Function]}
                  rows={3}
                />
              </div>
              <button
                className="stepone-next-btn inactive"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Send for Approval
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Approvals] ApprovalPopup should render component to show approver submit values 1`] = `
<div
  className="sliding-panel-container"
>
  <div
    className="sliding-panel"
  >
    <div
      className="title-row opp-header-text"
    >
      <button
        className="mob-steps-back"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Back"
          src="chevron-left-black.svg"
        />
      </button>
      Edit Opportunity
      <button
        className="panel-close-icon link-anchor-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Close"
          src="close.svg"
        />
      </button>
    </div>
    <div
      className="all-opportunity-steps-container"
    >
      <div
        className="opportunity-forms"
      >
        <div
          className=""
        >
          <div
            className="steps-one-forms"
          >
            <div
              className="stage-level-container"
            >
              <div
                className="stage-level-section"
              >
                <div
                  className="stage-row approval-data-rows"
                >
                  <div
                    className="form-group oppty-form-elements"
                  >
                    <label
                      className="cust-label"
                      htmlFor="comment"
                    >
                      Comment
                    </label>
                    <textarea
                      className="form-control"
                      id="comment"
                      onChange={[Function]}
                      placeholder="Add comment here"
                    />
                  </div>
                  <div
                    className="step-nextbtn-with-arrow stepsone-nxtbtn"
                  >
                    <button
                      className="approval-btn"
                      onClick={[Function]}
                      type="button"
                    >
                      Approve
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Approvals] ApprovalPopup should renders correctly 1`] = `
<div
  className="sliding-panel-container"
>
  <div
    className="sliding-panel"
  >
    <div
      className="title-row opp-header-text"
    >
      <button
        className="mob-steps-back"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Back"
          src="chevron-left-black.svg"
        />
      </button>
      Edit Opportunity
      <button
        className="panel-close-icon link-anchor-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Close"
          src="close.svg"
        />
      </button>
    </div>
    <div
      className="all-opportunity-steps-container"
    >
      <div
        className="opportunity-forms"
      >
        <div
          className=""
        >
          <div
            className="steps-one-forms"
          >
            <div
              className="stage-level-container"
            >
              <div
                className="stage-level-section"
              >
                <div
                  className="stage-row approval-data-rows"
                >
                  <div
                    className="form-group oppty-form-elements"
                  >
                    <label
                      className="cust-label"
                      htmlFor="comment"
                    >
                      Comment
                    </label>
                    <textarea
                      className="form-control"
                      id="comment"
                      onChange={[Function]}
                      placeholder="Add comment here"
                    />
                  </div>
                  <div
                    className="step-nextbtn-with-arrow stepsone-nxtbtn"
                  >
                    <button
                      className="approval-btn"
                      onClick={[Function]}
                      type="button"
                    >
                      Approve
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
